src/server/game/AI/SmartScripts/SmartScript.cpp

```diff
@@ -2337,6 +2337,44 @@ void SmartScript::ProcessAction(SmartScriptHolder& e, Unit* unit, uint32 var0, u
             }
             break;
         }
+        case SMART_ACTION_RANDOM_SOUND:
+        {
+            uint32 sounds[SMART_ACTION_PARAM_COUNT - 1];
+            sounds[0] = e.action.randomSound.sound1;
+            sounds[1] = e.action.randomSound.sound2;
+            sounds[2] = e.action.randomSound.sound3;
+            sounds[3] = e.action.randomSound.sound4;
+            sounds[4] = e.action.randomSound.sound5;
+            uint32 temp[SMART_ACTION_PARAM_COUNT - 1];
+            uint32 count = 0;
```

Comment:
Add a bool for OnlySelf, everything that is not 0 and 1 should default to 0. Else, the variable makes little sense.